### ISSUE TITLE
wish list: Janet's scripting syntax for gauche perhaps? (i.e. jpm install sh)

### ISSUE THREAD

**Issue by reuleaux** [shirok/Gauche#1002]
This is a wish list item (not a bug):
I really like _Janet's_ syntax for scripting (a DSL i.e.), in steps: install/read about [Janet](https://janet-lang.org/), and then, with the Janet package manager: jpm, installed with
```
make install-jpm-git
```
install _sh_ now:
```
.../jpm install sh
```
Find further info/examples about Janet (and sh) in [Janet for Mortals](https://janet.guide/), and in particular in the [Scripting chapter ](https://janet.guide/scripting/) there: ...some links to the 
[original work](https://github.com/andrewchambers/janet-sh) by Andrew Chambers, and follow the link to his  [blog post](https://acha.ninja/blog/dsl_for_shell_scripting/) from there (and by the way: don't confuse this _sh_ package, with the other one in [spork/sh)](https://janet-lang.org/api/spork/sh.html)

Maybe such a scripting DSL could be implemented in Gauche as well?
(it makes scripting in Janet a joy!). - Just an idea. - not that I have that much
time to implement it myself :-( 

Thanks. -Andreas

**Comment by reuleaux** [comment]
Hm, now that I take a closer look again (not having used gauche for a while),
I see that _do-process_ is quite capable. - So my bad (and sorry for the noise),
if this kind of writing scripts in a simple manner is already possible in gauche.
I do think however, that _janet_ is quite interesting a language.

**Comment by shirok** [comment]
Yeah, I think `gauche.process` has enough features, but I've been putting off writing a DSL layer for succinct shell-scripting.  Scsh has a kind of that, but I'm not very convinced it's ideal.  I'll check Janet out.

**Comment by reuleaux** [comment]
Great! - And I will some comments about _Janet_ some time (when I have a little more time - and when you have had a look): things that I like about
_Janet_ (some fresh approach to _Lisp_ ...), and things that I don't like so much (things that  are better in your _Gauche_, as I think) - which makes me think about "my ideal (_Lisp_-) language": a mixture of _Janet_ & _Gauche_ actually, written not in _C_ but in _Hare_ - but I doubt I will have the time to start such a project (well: maybe some time - this would be interesting in any case).
